src/deviceOauthService.ts

```diff
@@ -47,7 +53,15 @@ async function wait(ms = 1000): Promise<void> {
 
 async function makeRequest<T extends JsonMap>(options: HttpRequest): Promise<T> {
   const rawResponse = await new Transport().httpRequest(options);
-  const response = parseJsonMap<T>(rawResponse.body);
+  let response: T;
+  if (rawResponse?.headers && rawResponse.headers['content-type'] === 'text/html') {
+    response = {
```

Comment:
you could throw in the case of html error, rather than try to shoehorn it into T but with an error/error_description to cause it to throw on line 70.

as written, it would still barf the whole html page to the user (html => error_description => SfError message => final `throw` from `poll`).

When I did the jsforce fix for this in http-api, I put them html content to the logs.  You could, if throwing an SfError, put it in the `data` property if you think users might want it when using `--json`.